wordpress-mobile/WordPress-iOS
# UL&S: VoiceOver support for site address flow

...and UnifiedSignUpViewController

Ref. #14009 
Ref. https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/issues/358
Auth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/361

This PR updates three views to support VoiceOver:
- SiteAddressViewController
- SiteCredentialsViewController
- UnifiedSignUpViewController

And fixes bugs in the error handling that were only apparent after VoiceOver testing.

### To test the Site Address flow
1. Enable the `.unifiedSiteAddress` feature flag
2. `rake dependencies`
3. Build and run on a device
4. `Hey Siri turn on VoiceOver`
5. Log In > Or by entering your site address. 
6. It should read: "Site Address. Textfield. Is editing. example.com. Character mode. Insertion point at start."
7. (You can `hey siring turn off VoiceOver` so you can type in the address URL faster). Enter an invalid site, such as: fb.com (turn VoiceOver on again). Continue. Expected: the VoiceOver focus changes and reads the error message.
8. Enter a valid site. Select the site URL field again. It should change the reading now that a URL is entered. Tap Continue.
9. "Username. Textfield. Is editing. username. Character mode. Insertion point at start." Enter a username and tap the textfield. It will now read out "Username. Textfield. <your typed in username>. Character mode. Insertion point at end." Tap Next on the keyboard to move to the password field.
10. "Password. Secure textfield. Is editing. password. Character mode. Insertion point at start." Feel free to toggle the show / hide password button and hear the state read back to you. 

### To test the first Unified Sign Up screen
1. Enable the `.unifiedSignup` feature flag
2. Build and run on a device
3. Sign up with WordPress.com > Sign up with Email.
6. It should read: "Back button." and stop. This is expected behavior for navigating a screen with no textfields. Two-finger swipe down to listen to all of the UI. It should end on "send link by email".

PR submission checklist:
